packages/service-library/src/servicelib/background_task.py

```diff
@@ -99,9 +128,21 @@ async def stop_periodic_task(
     with log_context(
         logger,
         logging.DEBUG,
-        msg=f"cancel periodic background task '{asyncio_task.get_name()}'",
+        msg=f"stop periodic background task '{asyncio_task.get_name()}'",
     ):
-        await cancel_task(asyncio_task, timeout=timeout)
+        asyncio_task = cast(_ExtendedTask, asyncio_task)
+        continue_condition: ContinueCondition | None = asyncio_task.continue_condition
+        if continue_condition:
+            continue_condition.stop()
```

Comment:
ok what is the rationale here?
I see some major differences here:
- cancel directly cancels a task vs stop needs to wait for an iteration to complete (an iteration where we have no clue how long it takes), 
- therefore now you wait 2x the timeout in the worst case, how are the services coping with this did you check? I think we do not have the test anymore that would check starting/stopping services to see if they return a 0 exit code